portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -45,4 +42,24 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     response.setContentType("application/json");
     response.getWriter().println(json);
   }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get body of comment
+    String newComment = getParameter(request, "comment-body", "");
```

Comment:
Not part of the assignment, but still a good thought exercise. What will happen if the user clears comment-body text area and hits submit?

Reply:
It should put the empty comment in the array according to the code, but that should result in a warning or message to the user that it must be filled out! Now how to fix that...

Reply:
It usually depends on the application, but in this particular case I think posting empty comment is invalid scenario. I'd say it shouldn't even reach the server. Ideally there should be some code on the client that only posts when comment is non-empty. I'm pretty sure there is a solution for that (typical problems should have standard solutions), but I don't know the solution, will need to research it a bit.

Reply:
Gotcha, I am currently looking into a javascript function that verifies required elements have some value before reaching the backend.

Reply:
https://developer.mozilla.org/en-US/docs/Learn/Forms/Form_validation

Reply:
Oh this is actually really handy. HTML5 has built-in things to deal with this.